### PR TITLE
Emit auth error for wrong password

### PR DESCRIPTION
**Pull request by luin**
#164

### CODE REVIEW

File: lib/redis/event_handler.js [hunk 1]
```diff
@@ -12,9 +12,9 @@ exports.connectHandler = function (self) {
 
     // AUTH command should be processed before any other commands
     if (self.condition.auth) {
-      self.auth(self.condition.auth, function (err, res) {
-        if (err && err.message.match('no password is set')) {
-          console.warn('`auth` is specified in the client but not in the server.');
+      self.auth(self.condition.auth, function (err) {
+        if (err) {
+          self.emit('auth error', err);
```
Comment (naugtur) [inline lib/redis/event_handler.js]: could it be `auth-error` or `authError` or `authfailed` ?

Comment (luin) [inline lib/redis/event_handler.js]: `authError` looks good to me

**Comment by naugtur** [comment]
That doesn't look like a change that'd make all tests fail...

**Comment by luin** [comment]
The test fails because it expects ioredis would print a warning message if the server doesn't require a password: https://github.com/luin/ioredis/blob/master/test/functional/auth.js#L44. I'm thinking whether we can remove this test.

**Comment by naugtur** [comment]
Now that the user is in control documenting the auth error event should be enough. 
For an app that doesn't care about error handling ignoring this seems like the way to go ;)

**Comment by luin** [comment]
Released with 1.9.1 :rocket:

**Comment by naugtur** [comment]
:+1:

**Pull request merged**